crm_lead_to_event_registration/models/crm_lead.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    event_registration_id = fields.Many2one(
+        "event.registration",
+        "Event registration",
+        readonly=True,
+        help="Registration generated by this lead/opportunity.")
+
+    @api.multi
+    def action_generate_event_registration(self, event):
+        """Generate an event registration."""
+        er = self.env["event.registration"]
+        for s in self:
+            s.event_registration_id = er.create({
+                "event_id": event.id,
+                "partner_id": s.partner_id.id,
+                "nb_register": 1,
+                "name": s.contact_name,
+                "email": s.email_from,
+                "phone": s.phone,
+            })
+            # Load data from partner if available
+            s.event_registration_id._onchange_partner()
+
+    @api.multi
+    @api.constrains("probability", "stage_id")
+    def _check_status_confirm_registration(self):
+        """If the opportunity is won/lost, open/cancel registration."""
+        won = self._track["stage_id"]["crm.mt_lead_won"]
+        lost = self._track["stage_id"]["crm.mt_lead_lost"]
+        for s in self:
+            if won(s, s.env.cr, s.env.uid, s, s.env.context):
+                s.event_registration_id.registration_open()
```

Comment:
Why do you execute this action if this is a constrain. Constrain was designed to raise ValidationError when some conditions are triggered and return True if data is valid.
I'm not sure what could happen if any ValidationError is raised after action execute

IMHO, this action could be in write method